test-templates/src/groups.rs

```diff
@@ -111,6 +111,18 @@ macro_rules! __test_group {
                         assert_eq!(context.mul_with_table(&bad_table, &b), None);
                     }
                 }
+
+                // The table size is only determined by the number of non-zero entries.
```

Comment:
I think the table size could be anything the user sets.
E.g. [here](https://github.com/arkworks-rs/poly-commit/blob/f6a0c13bf44d7d1776f238bc9afabeab3c5a9511/poly-commit/src/multilinear_pc/mod.rs#L58) the table size is `num_vars`, while we have `2^(num_vars+1) - 1` scalars, if I'm reading the code correctly.

Reply:
In that case I'll rename `max_num_scalars` to just `num_scalars`.